### PR TITLE
Contain temp dirs in 'tmp' dir, so cwd doesn't clutter with uuid-named directories

### PR DESCRIPTION
**Pull request by stuarthicks**

### CODE REVIEW

File: gokay.go [hunk 1]
```diff
@@ -49,12 +49,14 @@ func main() {
...
-	if err := os.Mkdir(tempDir, os.ModePerm); err != nil {
-		log.Fatalf("Error creating directory %v: %v\n", tempDir, err)
-	}
+	tempDir := fmt.Sprintf("%s/tmp/%s", fileDir, tempName)
 	tempFile := fmt.Sprintf("%s/%s.go", tempDir, tempName)
 
+	// os.Mkdir errors when the dir already exists, which is fine.
+	// Real errors will still be detected when we try use the actual tmpfile.
+	_ = os.Mkdir(fileDir+"/tmp", os.ModePerm)
+	_ = os.Mkdir(tempDir, os.ModePerm)
```
Comment (stuarthicks) [inline gokay.go]: Happy to put non-fatal logging back here if necessary, although imo it's not useful information.

File: gokay.go [hunk 2]
```diff
@@ -109,8 +111,6 @@ func main() {
 
 	fmt.Fprintf(outWriter, "}\n")
 
-	tempOut.Close()
```
Comment (stuarthicks) [inline gokay.go]: This was already called in a defer earlier. No need to call it again.

File: gokay.go [hunk 3]
```diff
@@ -49,12 +49,14 @@ func main() {
...
-	tempDir := fmt.Sprintf("%s/%s", fileDir, tempName)
-	if err := os.Mkdir(tempDir, os.ModePerm); err != nil {
-		log.Fatalf("Error creating directory %v: %v\n", tempDir, err)
-	}
+	tempDir := fmt.Sprintf("%s/tmp/%s", fileDir, tempName)
 	tempFile := fmt.Sprintf("%s/%s.go", tempDir, tempName)
 
+	// os.Mkdir errors when the dir already exists, which is fine.
+	// Real errors will still be detected when we try use the actual tmpfile.
+	_ = os.Mkdir(fileDir+"/tmp", os.ModePerm)
```
Comment (soldiermoth) [inline gokay.go]: Should be able to drop the `_ = `:
```go
os.Mkdir(fileDir+"/tmp", os.ModePerm)
os.Mkdir(tempDir, os.ModePerm)
```

Comment (stuarthicks) [inline gokay.go]: Yep. Done.

**Pull request merged**